internal/app/configuration/proxyconfig.go

```diff
@@ -1,38 +1,30 @@
 package configuration
 
 import (
-	"fmt"
-	"net/url"
+	"context"
 
 	"github.com/form3tech-oss/pact-proxy/internal/app/pactproxy"
+	"github.com/sethvargo/go-envconfig"
+	log "github.com/sirupsen/logrus"
 )
 
-type ProxyConfig struct {
-	ServerAddress string
-	Target        string
-}
+func NewFromEnv() pactproxy.Config {
+	ctx := context.Background()
 
-func ConfigureProxy(config ProxyConfig) error {
-	targetURL, err := url.Parse(config.Target)
+	var c pactproxy.Config
+	err := envconfig.Process(ctx, &c)
 	if err != nil {
-		return err
-	}
-
-	serverAddr := config.ServerAddress
-	if serverAddr == "" {
-		serverAddr = fmt.Sprintf("http://:%s", targetURL.Port())
-	}
-
-	serverAddrURL, err := url.Parse(serverAddr)
-	if err != nil {
-		return err
+		log.Fatal(err.Error())
```

Comment:
I would prefer if the function returned `pactproxy.Config, error` and we did the fatal in `main.go`.